classes/class-dintero-checkout-meta-box.php

```diff
@@ -81,8 +81,13 @@ public function print_content( $dintero_order ) {
 		$account_id     = trim( get_option( 'woocommerce_dintero_checkout_settings', array( 'account_id' => '' ) )['account_id'] );
 		$transaction_id = $order->get_meta( '_dintero_transaction_id' );
 
-		/* Remove duplicate words from the payment method type (e.g., swish.swish → Swish). Otherwise, prints as is (e.g., collector.invoice → Collector Invoice). */
-		$payment_method = implode( ' ', array_unique( explode( ' ', ucwords( str_replace( '.', ' ', $dintero_order['payment_product_type'] ) ) ) ) );
+		$payment_method = $order->get_meta( '_dintero_payment_method' );
+		if ( empty( $payment_method ) ) {
+			/* Remove duplicate words from the payment method type (e.g., swish.swish → Swish). Otherwise, prints as is (e.g., collector.invoice → Collector Invoice). */
+			$payment_method = implode( ' ', array_unique( explode( ' ', ucwords( str_replace( '.', ' ', $dintero_order['payment_product_type'] ) ) ) ) );
```

Comment:
You have this same logic in two places now, where the only difference is the value passed to it.

I suggest making a function for this to get the payment method from a passed string, that way if any logic needs to change its only in one place.
Also i suggest to more easily maintain this in the future to break this up a bit instead of having it as a one liner. It would make it a bit easier to follow in the code what is happening.
With that you can also do the explode and array unique first, then upper case the first letter and lastly the implode. That way you don't have to run ucwords on strings that will get removed anyway.

Reply:
The other place you have this is `includes/dintero-checkout-functions.php::169`